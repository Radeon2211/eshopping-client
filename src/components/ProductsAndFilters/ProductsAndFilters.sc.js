import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ProductsTopbar = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-bottom: ${({ theme }) => theme.spacings.level3};
  padding: 0 ${({ theme }) => theme.spacings.level3} 0 0;

  @media only screen and (max-width: 37.5em) {
    padding: 0;
  }
`;

export const ProductsBottombar = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin-top: ${({ theme }) => theme.spacings.level3};
  padding: 0 ${({ theme }) => theme.spacings.level3} 0 0;

  @media only screen and (max-width: 37.5em) {
    padding: 0;
  }
`;
