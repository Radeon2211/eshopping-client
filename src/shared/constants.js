import * as Yup from 'yup';
import { getCountries } from 'country-fns';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productConditions = {
  NEW: 'new',
  USED: 'used',
  NOT_APPLICABLE: 'not_applicable',
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([productConditions.NEW, productConditions.USED, productConditions.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
};

export const listOfAreaCodes = getCountries().map(({ name, dial }) => {
  const finalValue = `+${dial} ${name.split('(')[0].trim()}`;
  return {
    value: dial,
    label: finalValue,
  };
});

export const listOfCountries = getCountries().map(({ name }) => {
  const finalValue = name.split('(')[0].trim();
  return {
    value: finalValue,
    label: finalValue,
  };
});

export const productPages = {
  ALL_PRODUCTS: 'ALL_PRODUCTS',
  MY_PRODUCTS: 'MY_PRODUCTS',
  USER_PRODUCTS: 'USER_PRODUCTS',
};

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const orderTypes = {
  PLACED_ORDERS: 'PLACED_ORDERS',
  SELL_HISTORY: 'SELL_HISTORY',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const modalTypes = {
  ABOUT_WEBSITE: 'ABOUT_WEBSITE',
  ADD_ADMIN: 'ADD_ADMIN',
  ADD_PRODUCT: 'ADD_PRODUCT',
  BUY_PRODUCTS: 'BUY_PRODUCTS',
  CART_ITEM_ADDED: 'CART_ITEM_ADDED',
  CHANGE_NAME: 'CHANGE_NAME',
  CHANGE_EMAIL: 'CHANGE_EMAIL',
  CHANGE_USERNAME: 'CHANGE_USERNAME',
  CHANGE_PHONE_NUMBER: 'CHANGE_PHONE_NUMBER',
  CHANGE_ADDRESS: 'CHANGE_ADDRESS',
  CHANGE_PASSWORD: 'CHANGE_PASSWORD',
  CHANGE_CONTACTS: 'CHANGE_CONTACTS',
  CHANGE_DELIVERY_ADDRESS: 'CHANGE_DELIVERY_ADDRESS',
  CLEAR_CART: 'CLEAR_CART',
  DELETE_ACCOUNT: 'DELETE_ACCOUNT',
  DELETE_PRODUCT: 'DELETE_PRODUCT',
  EDIT_PRODUCT: 'EDIT_PRODUCT',
  LOGIN: 'LOGIN',
  REMOVE_ADMIN: 'REMOVE_ADMIN',
  RESET_PASSWORD: 'RESET_PASSWORD',
  PENDING_USER_INFO: 'PENDING_USER_INFO',
  SEND_VERIFICATION_LINK: 'SEND_VERIFICATION_LINK',
  SIGNUP: 'SIGNUP',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: `Price - ascending` },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: `Price - ascending` },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const updateCartActions = {
  INCREMENT: 'INCREMENT',
  DECREMENT: 'DECREMENT',
  NUMBER: 'NUMBER',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};

export const userStatuses = {
  ACTIVE: 'active',
  PENDING: 'pending',
};
