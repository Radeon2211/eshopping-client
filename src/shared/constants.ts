import * as Yup from 'yup';
import countryCodes from 'country-codes-list';
import { ProductCondition, ProductPhotoFieldValue } from './types/enums';

export const anyStringRule = Yup.string().min(1).required();

export const userRules = {
  email: Yup.string().email().max(320).trim().required(),
  hideEmail: Yup.bool(),
  username: Yup.string().min(3).max(20).trim().required(),
  password: Yup.string().min(7).max(64).trim().required(),
  firstName: Yup.string().max(60).trim().required(),
  lastName: Yup.string().max(80).trim().required(),
  phonePrefix: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
  phoneNumber: Yup.string()
    .min(5)
    .max(15)
    .matches(/^[\d-]{5,15}$/)
    .trim()
    .required(),
  hidePhone: Yup.bool(),
  street: Yup.string().max(60).trim().required(),
  zipCode: Yup.string().max(12).trim().required(),
  city: Yup.string().max(100).trim().required(),
  country: Yup.object()
    .shape({
      value: Yup.string().required(),
      label: Yup.string().required(),
    })
    .nullable()
    .required(),
};

export const productRules = {
  name: Yup.string().max(150).trim().required(),
  price: Yup.number().min(0.01).max(1000000).required(),
  quantity: Yup.number().min(1).max(100000).required(),
  condition: Yup.string()
    .oneOf([ProductCondition.NEW, ProductCondition.USED, ProductCondition.NOT_APPLICABLE])
    .required(),
  description: Yup.string().max(600).trim(),
  photo: Yup.mixed()
    .test('is-error', (value) => value !== ProductPhotoFieldValue.ERROR)
    .nullable(),
};

export const listOfAreaCodes = countryCodes.customArray({
  value: '{countryCallingCode}',
  label: '+{countryCallingCode} {countryNameEn}',
});

export const listOfCountries = countryCodes.customArray({
  value: '{countryNameEn}',
  label: '{countryNameEn}',
});

export const itemTypes = {
  CART: 'CART',
  ORDER: 'ORDER',
  TRANSACTION: 'TRANSACTION',
};

export const listItemTypes = {
  PRODUCT: 'PRODUCT',
  ORDER: 'ORDER',
};

export const inputKinds = {
  INPUT: 'INPUT',
  TEXTAREA: 'TEXTAREA',
  SELECT: 'SELECT',
  RADIO: 'RADIO',
};

export const defaultProductsPerPage = 10;

export const defaultOrdersPerPage = 6;

export const defaultAppPath = '/products?p=1';

export const filtersActions = {
  INIT_STATE: 'INIT_STATE',
  SET_SORT_BY: 'SET_SORT_BY',
  SET_MIN_PRICE: 'SET_MIN_PRICE',
  SET_MAX_PRICE: 'SET_MAX_PRICE',
  SET_CONDITION: 'SET_CONDITION',
};

export const sliderPositionsActions = {
  SET_BOTH: 'SET_BOTH',
  SET_LEFT: 'SET_LEFT',
  SET_RIGHT: 'SET_RIGHT',
};

export const sortProductsOptions = [
  { value: undefined, label: 'Default sorting' },
  { value: 'price:asc', label: 'Price - ascending' },
  { value: 'price:desc', label: 'Price - descending' },
  { value: 'name:asc', label: 'Name - A to Z' },
  { value: 'name:desc', label: 'Name - Z to A' },
];

export const sortOrdersOptions = [
  { value: 'createdAt:desc', label: 'Date - from newest' },
  { value: 'createdAt:asc', label: 'Date - from oldest' },
  { value: 'overallPrice:asc', label: 'Price - ascending' },
  { value: 'overallPrice:desc', label: 'Price - descending' },
];

export const singleInfoNames = {
  USERNAME: 'Username',
  NAME: 'Name',
  EMAIL: 'Email',
  PHONE_NUMBER: 'Phone number',
  ADDRESS: 'Address',
  CONTACTS: 'Contacts',
};

export const defaultErrorMessage = 'Something went wrong';

export const productsPerPageControllerOptions = [
  { value: 1, label: '1' },
  { value: 2, label: '2' },
  { value: 3, label: '3' },
  { value: 4, label: '4' },
  { value: 5, label: '5' },
  { value: 10, label: '10' },
  { value: 15, label: '15' },
  { value: 20, label: '20' },
  { value: 25, label: '25' },
];

export const defaultScrollToConfig: ScrollToOptions = {
  top: 0,
  left: 0,
  behavior: 'smooth',
};
