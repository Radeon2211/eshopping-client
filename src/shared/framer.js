import theme from '../styled/theme';

export const stepFormVariants = {
  hidden: {
    opacity: 0,
  },
  visible: {
    opacity: 1,
    transition: { duration: theme.durations.level1, delay: theme.durations.level1 },
  },
};

export const backdropVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    transition: { duration: theme.durations.level3 },
  },
};

export const modalVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
  },
  visible: {
    opacity: 1,
    transition: { delay: theme.durations.level2 },
    pointerEvents: 'initial',
  },
};

export const messageBoxVariants = {
  hidden: {
    pointerEvents: 'none',
    y: '-150%',
  },
  visible: {
    transition: { type: 'spring', duration: theme.durations.level2, stiffness: 100 },
    pointerEvents: 'initial',
    y: 0,
  },
};

export const dropdownVariants = {
  hidden: {
    opacity: 0,
    pointerEvents: 'none',
    y: '-10%',
  },
  visible: {
    opacity: 1,
    pointerEvents: 'initial',
    y: '0',
  },
};
