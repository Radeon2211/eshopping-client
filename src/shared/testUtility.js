// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
