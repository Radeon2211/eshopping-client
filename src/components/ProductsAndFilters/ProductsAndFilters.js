import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { useWindowWidth } from '@react-hook/window-size';
import * as SC from './ProductsAndFilters.sc';
import SideBySide from '../UI/SideBySide';
import Filters from '../Filters/Filters';
import ProductList from '../ProductList/ProductList';
import Panel from '../UI/Panel';
import InputPagination from '../Pagination/InputPagination/InputPagination';
import NumberPagination from '../Pagination/NumberPagination/NumberPagination';
import PaginationCounter from '../Pagination/PaginationCounter/PaginationCounter';
import QuantityPerPageController from '../Pagination/QuantityPerPageController';
import { listItemTypes } from '../../shared/constants';
import MyIcon from '../UI/MyIcon';
import { ReactComponent as FiltersIcon } from '../../images/SVG/filters.svg';
import { ReactComponent as ArrowIcon } from '../../images/SVG/arrow.svg';

const ProductsAndFilters = (props) => {
  const { page } = props;

  const [filtersIsVisible, setFiltersIsVisible] = useState(false);
  const windowWidth = useWindowWidth();

  const products = useSelector((state) => state.product.products);
  const productCount = useSelector((state) => state.product.productCount);
  const isDataLoading = useSelector((state) => state.ui.isDataLoading);
  const maxQuantityPerPage = useSelector((state) => state.ui.maxQuantityPerPage);

  let inputPagination = null;
  let numberPagination = null;
  if (productCount !== 0) {
    inputPagination = (
      <SC.ProductsTopbar>
        <QuantityPerPageController maxQuantityPerPage={maxQuantityPerPage} />
        <InputPagination
          itemQuantity={productCount}
          isDataLoading={isDataLoading}
          maxQuantityPerPage={maxQuantityPerPage}
        />
      </SC.ProductsTopbar>
    );
    numberPagination = (
      <SC.ProductsBottombar>
        <PaginationCounter
          itemQuantity={productCount}
          itemsType={listItemTypes.PRODUCT}
          maxQuantityPerPage={maxQuantityPerPage}
        />
        <NumberPagination
          itemQuantity={productCount}
          isDataLoading={isDataLoading}
          maxQuantityPerPage={maxQuantityPerPage}
        />
      </SC.ProductsBottombar>
    );
  }
  let filtersToggler = null;
  if (windowWidth <= 1200) {
    filtersToggler = (
      <SC.FiltersToggler onClick={() => setFiltersIsVisible((prevState) => !prevState)}>
        <MyIcon size="small">
          <FiltersIcon />
        </MyIcon>
        <span className="label">Filters</span>
        <MyIcon size="small" rotation={filtersIsVisible ? -90 : 90}>
          <ArrowIcon />
        </MyIcon>
      </SC.FiltersToggler>
    );
  }

  return (
    <SC.Wrapper>
      {filtersToggler}
      <SideBySide proportion="1/3" makeVerticalWhen={1200}>
        <Filters
          products={products}
          isDataLoading={isDataLoading}
          isVisible={filtersIsVisible || windowWidth > 1200}
        />
        <Panel>
          {inputPagination}
          <ProductList products={products} isDataLoading={isDataLoading} page={page} />
          {numberPagination}
        </Panel>
      </SideBySide>
    </SC.Wrapper>
  );
};

ProductsAndFilters.propTypes = {
  page: PropTypes.string.isRequired,
};

export default ProductsAndFilters;
