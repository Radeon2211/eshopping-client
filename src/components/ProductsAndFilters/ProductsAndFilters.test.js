import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import matchMediaPolyfill from 'mq-polyfill';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ProductsAndFilters from './ProductsAndFilters';
import { createProductItem, renderAppPart } from '../../shared/testUtility/testUtility';
import { productPages, defaultProductsPerPage } from '../../shared/constants';

const mockStore = configureMockStore([thunk]);

const createStore = (products, productCount, minPrice, maxPrice) =>
  mockStore({
    product: {
      products,
      productCount,
      minPrice,
      maxPrice,
    },
    ui: {
      isDataLoading: false,
      productsPerPage: defaultProductsPerPage,
    },
  });

const defaultProducts = [
  createProductItem({
    id: 'p1',
    sellerUsername: 'user1',
    price: 10.6,
    quantity: 4,
    name: 'product1',
  }),
  createProductItem({
    id: 'p2',
    sellerUsername: 'user1',
    price: 299.98,
    quantity: 6,
    name: 'product2',
  }),
];

const setUp = (store, search = '?p=1') => {
  return renderAppPart(<ProductsAndFilters page={productPages.ALL_PRODUCTS} />, {
    pathname: '/products',
    search,
    store,
  });
};

beforeAll(() => {
  matchMediaPolyfill(window);
  window.resizeTo = function resizeTo(width, height) {
    Object.assign(this, {
      innerWidth: width,
      innerHeight: height,
    }).dispatchEvent(new this.Event('resize'));
  };
});

describe('<ProductsAndFilters />', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  describe('check how renders', () => {
    it('should render everything correctly with two products', () => {
      const store = createStore(defaultProducts, defaultProducts.length, 10.6, 299.98);
      window.resizeTo(1201, 800);
      setUp(store);
      expect(screen.getByTestId('Filters'));
      expect(screen.getByTestId('ProductsAndFilters-product-list-section'));
      expect(screen.getAllByTestId('ProductItem')).toHaveLength(2);
    });

    it('should render only <ProductList /> if product array is empty', () => {
      const store = createStore([], 0);
      window.resizeTo(1199, 800);
      setUp(store);
      expect(screen.queryByTestId('Filters')).not.toBeInTheDocument();
      expect(
        screen.queryByTestId('ProductsAndFilters-product-list-section'),
      ).not.toBeInTheDocument();
      expect(screen.getByTestId('ProductList'));
    });

    it('should render <Filters /> if at least two params (except p)', () => {
      const store = createStore([], 0);
      window.resizeTo(1201, 800);
      setUp(store, '?p=1&name=testName&minPrice=10');
      expect(screen.getByTestId('Filters'));
    });

    it('should render <Filters /> if one param (except p)', () => {
      const store = createStore([], 0);
      window.resizeTo(1201, 800);
      setUp(store, '?p=1&minPrice=10');
      expect(screen.getByTestId('Filters'));
    });

    it('should NOT render product list section if products are undefined', () => {
      const store = createStore(undefined, 2);
      setUp(store);
      expect(
        screen.queryByTestId('ProductsAndFilters-product-list-section'),
      ).not.toBeInTheDocument();
    });
  });
});
