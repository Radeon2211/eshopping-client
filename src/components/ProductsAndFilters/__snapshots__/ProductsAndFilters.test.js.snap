// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsAndFilters /> Check how renders should render everything correctly with two products 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ eOpHEH"
  >
    <section
      class="sc-jSgupP ikrxvH"
      data-testid="Filters"
    >
      <div
        class="sc-gsTCUz giqxPL"
      >
        <div
          class="select css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-yk16xz-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
            >
              <div
                class=" css-1uccc91-singleValue"
              >
                Default sorting
              </div>
              <input
                aria-autocomplete="list"
                class="css-wmatm6-dummyInput-DummyInput"
                id="react-select-3-input"
                readonly=""
                tabindex="0"
                value=""
              />
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <input
            name="sortBy"
            type="hidden"
            value=""
          />
        </div>
        <div
          class="sc-kEjbxe dWwZfv"
        >
          <div
            class="sc-iqHYGH efDrOl"
          >
            <span
              class="sc-gKsewC ISnTb"
              color=""
              display="inline-block"
              overflow="visible"
              size="3"
              spacing="0px"
              transform=""
            >
              Price ($)
            </span>
            <div
              class="inputs"
            >
              <input
                class="sc-dQppl coboiF"
                data-testid="NumberInput-minPrice"
                name="minPrice"
                type="number"
                value="10.6"
              />
              <span
                class="inputs-gap"
              >
                â€”
              </span>
              <input
                class="sc-dQppl coboiF"
                data-testid="NumberInput-maxPrice"
                name="maxPrice"
                type="number"
                value="299.98"
              />
            </div>
          </div>
          <div
            class="sc-crrsfI cERfvJ"
          >
            <input
              class="input min"
              data-testid="PriceSlider-price-range-min"
              max="299.98"
              min="10.6"
              name="minPrice"
              step="0.29998"
              type="range"
              value="10.6"
            />
            <input
              class="input max"
              data-testid="PriceSlider-price-range-max"
              max="299.98"
              min="10.6"
              name="maxPrice"
              step="0.29998"
              type="range"
              value="299.98"
            />
            <div
              class="track"
            />
            <div
              class="range"
            />
            <div
              class="thumb left"
            />
            <div
              class="thumb right"
            />
          </div>
        </div>
        <div
          class="sc-hKgILt ifwnTa"
        >
          <span
            class="sc-gKsewC ISnTb"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            Condition
          </span>
          <div
            class="sc-eCssSg euCWRc"
          >
            <input
              data-testid="Filters-checkbox-new"
              id="new"
              name="new"
              type="checkbox"
            />
            <label
              for="new"
            >
              new
            </label>
          </div>
          <div
            class="sc-eCssSg euCWRc"
          >
            <input
              data-testid="Filters-checkbox-used"
              id="used"
              name="used"
              type="checkbox"
            />
            <label
              for="used"
            >
              used
            </label>
          </div>
          <div
            class="sc-eCssSg euCWRc"
          >
            <input
              data-testid="Filters-checkbox-not-applicable"
              id="not_applicable"
              name="not_applicable"
              type="checkbox"
            />
            <label
              for="not_applicable"
            >
              not applicable
            </label>
          </div>
        </div>
        <button
          class="sc-iBPRYJ fDMRnL"
          color="blue"
          data-testid="Filters-submit-btn"
          type="button"
        >
          Filter
        </button>
      </div>
    </section>
    <section
      class="sc-jSgupP ikrxvH"
      data-testid="ProductsAndFilters-product-list-section"
    >
      <div
        class="sc-fKFyDc gMvzTj"
      >
        <div
          class="sc-dmlrTW fHfiAZ"
        >
          <label
            class="label"
            for="perPageQuantity"
          >
            Per page
          </label>
          <div
            class="select css-2b097c-container"
            id="perPageQuantity"
          >
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-l640om-control"
            >
              <div
                class=" css-3cb03z-ValueContainer"
              >
                <div
                  class=" css-rrw4q2-SingleValue"
                >
                  10
                </div>
                <input
                  aria-autocomplete="list"
                  class="css-wmatm6-dummyInput-DummyInput"
                  id="react-select-2-input"
                  readonly=""
                  tabindex="0"
                  value=""
                />
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-ml9fjp-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-hHftDr fbtuLP"
          data-testid="InputPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-left-arrow"
          >
            <div
              class="sc-bqyKva kNsZJh"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <form
            class="form-number"
          >
            <input
              class="sc-dQppl bJsXeH"
              data-testid="NumberInput-page"
              name="page"
              type="number"
              value="1"
            />
          </form>
          <span
            class="sc-gKsewC hpFlQD"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            of
          </span>
          <span
            class="sc-gKsewC hpFlQD"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="InputPagination-right-arrow"
          >
            <div
              class="sc-bqyKva fuAYNn"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
      <div
        class="sc-bkzZxe fwDVgA"
        data-testid="ProductList"
      >
        <a
          data-testid="ProductItem"
        >
          <div
            class="sc-kstrdz cJSdfS sc-hBEYos cBfiMf"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <div
              class="sc-kstrdz bnACqZ sc-fodVxV XQHig"
              data-testid="ProductThumbnail"
              direction="row"
              height="15"
              spacing=""
              width="13"
              wrap="nowrap"
            >
              <img
                alt="product1"
                data-testid="ProductThumbnail-img"
                src="no-photo.png"
              />
            </div>
            <div
              class="sc-kstrdz buzgIP"
              direction="column"
              spacing=""
              width="100%"
              wrap="nowrap"
            >
              <span
                class="sc-gKsewC ISnTb"
                color=""
                display="inline-block"
                overflow="visible"
                size="3"
                spacing="0px"
                transform=""
              >
                product1
              </span>
              <span
                class="sc-gKsewC oZHlp"
                color=""
                data-testid="ProductItem-condition"
                display="inline-block"
                overflow="visible"
                size="1"
                spacing="0px"
                transform=""
              >
                <span
                  class="sc-gKsewC bhnoUF"
                  color="#777"
                  display="inline-block"
                  overflow="visible"
                  spacing="0px"
                  transform=""
                >
                  Condition:
                </span>
                 New
              </span>
              <span
                class="sc-gKsewC kOxGPE"
                color=""
                data-testid="ProductItem-price"
                display="inline-block"
                overflow="visible"
                size="5"
                spacing="0px"
                transform=""
              >
                $10.60
              </span>
            </div>
          </div>
        </a>
        <a
          data-testid="ProductItem"
        >
          <div
            class="sc-kstrdz cJSdfS sc-hBEYos cBfiMf"
            direction="row"
            spacing="3"
            width="100%"
            wrap="nowrap"
          >
            <div
              class="sc-kstrdz bnACqZ sc-fodVxV XQHig"
              data-testid="ProductThumbnail"
              direction="row"
              height="15"
              spacing=""
              width="13"
              wrap="nowrap"
            >
              <img
                alt="product2"
                data-testid="ProductThumbnail-img"
                src="no-photo.png"
              />
            </div>
            <div
              class="sc-kstrdz buzgIP"
              direction="column"
              spacing=""
              width="100%"
              wrap="nowrap"
            >
              <span
                class="sc-gKsewC ISnTb"
                color=""
                display="inline-block"
                overflow="visible"
                size="3"
                spacing="0px"
                transform=""
              >
                product2
              </span>
              <span
                class="sc-gKsewC oZHlp"
                color=""
                data-testid="ProductItem-condition"
                display="inline-block"
                overflow="visible"
                size="1"
                spacing="0px"
                transform=""
              >
                <span
                  class="sc-gKsewC bhnoUF"
                  color="#777"
                  display="inline-block"
                  overflow="visible"
                  spacing="0px"
                  transform=""
                >
                  Condition:
                </span>
                 New
              </span>
              <span
                class="sc-gKsewC kOxGPE"
                color=""
                data-testid="ProductItem-price"
                display="inline-block"
                overflow="visible"
                size="5"
                spacing="0px"
                transform=""
              >
                $299.98
              </span>
            </div>
          </div>
        </a>
      </div>
      <div
        class="sc-dIUggk gVtHox"
      >
        <span
          class="sc-gKsewC rBmjh"
          color="#777"
          display="inline-block"
          overflow="visible"
          size="1"
          spacing="0px"
          transform=""
        >
          1 - 2 of 2 products
        </span>
        <div
          class="sc-idOhPF gwyQBe"
          data-testid="NumberPagination"
        >
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-left-arrow"
          >
            <div
              class="sc-bqyKva kNsZJh"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
          <a
            class="number-link active"
            data-testid="NumberPagination-page1"
          >
            1
          </a>
          <span
            class="of"
          >
            of
          </span>
          <span
            class="of"
          >
            1
          </span>
          <a
            class="arrow hide-arrow"
            data-testid="NumberPagination-right-arrow"
          >
            <div
              class="sc-bqyKva fuAYNn"
              color=""
            >
              <svg>
                arrow.svg
              </svg>
            </div>
          </a>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`<ProductsAndFilters /> Check how renders should render only <ProductList /> 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ eOpHEH"
  >
    <div
      class="sc-bkzZxe fwDVgA"
      data-testid="ProductList"
    >
      <h4
        class="sc-fFubgz kDCMhs"
      >
        We didn't find any matching results. Try to search something else
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
